ivy/functional/backends/paddle/activations.py

```diff
@@ -73,34 +60,30 @@ def gelu(
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
     if paddle.is_complex(x):
-        sqrt_2_over_pi = 0.7978845608
-        # the other magic number comes directly from the formula in
-        # https://doi.org/10.48550/arXiv.1606.08415
-        return (
-            0.5
-            * x
-            * (1 + paddle_backend.tanh(sqrt_2_over_pi * (x + 0.044715 * x * x * x)))
-        )
-    if x.dtype in unsupported_dtypes:
-        return F.gelu(x.cast("float32"), approximate=approximate).cast(x.dtype)
+        if approximate:
+            return (
+                0.5
+                * x
+                * (1 + paddle_backend.tanh(0.7978845608 * (x + 0.044715 * x * x * x)))
+            )
+        return 0.5 * x * (1 + paddle_backend.erf(x / paddle_backend.sqrt(2)))
     return F.gelu(x, approximate=approximate)
 
 
-@with_unsupported_device_and_dtypes(
-    {"2.5.1 and below": {"cpu": ("bfloat16",)}}, backend_version
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float16", "float32", "float64", "complex")},
```

Comment:
`float16` doesn't seem to be supported by the function

Reply:
yeah, the paddle docs are misleading, from now on I'll use brute force dtype checking